src/hooks/useNotifications.js

```diff
@@ -0,0 +1,42 @@
+import { useState, useEffect } from 'react'
+import messaging from '@react-native-firebase/messaging'
+
+import { useClient } from 'cozy-client'
+import { removeNotificationDeviceToken } from '/libs/client'
+import {
+  handleNotificationTokenReceiving,
+  handleNotificationOpening,
+  requestAndGetNotificationPermission
+} from '/libs/notifications/notifications'
+
+export const useNotifications = () => {
+  const client = useClient()
+
+  const [areNotificationsEnabled, setAreNotificationsEnabled] = useState(false)
+
+  useEffect(() => {
+    ;(async function () {
```

Comment:
While I agree with the spirit I think IIFE are not that great with the mandatory semi-colon prefix.
Best to name things a bit more even if it feels verbose, a good middle ground is more robust